packages/blade/src/components/Table/Table.web.tsx

```diff
@@ -0,0 +1,539 @@
+        tableComponentArray[0].props.children,
+      );
+      const tableHeader = tableComponentArrayChildren.find(
+        (child) => getComponentId(child) === ComponentIds.TableHeader,
+      );
+      const tableHeaderChildrenArray = React.isValidElement(tableHeader)
+        ? React.Children.toArray(tableHeader.props.children)
+        : null;
+      const tableHeaderRow = tableHeaderChildrenArray?.find(
+        (child) => getComponentId(child) === ComponentIds.TableHeaderRow,
+      );
+      const tableHeaderRowChildrenArray = React.isValidElement(tableHeaderRow)
+        ? React.Children.toArray(tableHeaderRow.props.children)
+        : null;
+      const tableHeaderCells = tableHeaderRowChildrenArray
+        ? tableHeaderRowChildrenArray.filter(
+            (child) => getComponentId(child) === ComponentIds.TableHeaderCell,
+          )
+        : null;
+      return tableHeaderCells?.length ?? 0;
+    }
+  }
+  return 0;
+};
+
+const StyledReactTable = styled(ReactTable)<{ styledProps?: { height?: BoxProps['height'] } }>(
+  ({ styledProps }) => {
+    const { theme } = useTheme();
+    const styledPropsCSSObject = getBaseBoxStyles({
+      theme,
+      height: styledProps?.height,
+    });
+
+    return {
+      '&&&': {
+        ...styledPropsCSSObject,
+      },
+    };
+  },
+);
+
+const Table = <Item,>({
+  children,
+  data,
+  selectionType,
+  onSelectionChange,
+  isHeaderSticky,
+  isFooterSticky,
+  isFirstColumnSticky,
+  rowDensity = 'normal',
+  onSortChange,
+  sortFunctions,
+  toolbar,
+  pagination,
+  height,
+  showStripes,
+  gridTemplateColumns,
+  surfaceLevel = 2,
+  isLoading = false,
+  isRefreshing = false,
+  ...styledProps
+}: TableProps<Item>): React.ReactElement => {
+  const { theme } = useTheme();
+  const [selectedRows, setSelectedRows] = React.useState<TableNode<unknown>['id'][]>([]);
+  const [disabledRows, setDisabledRows] = React.useState<TableNode<unknown>['id'][]>([]);
+  const [totalItems, setTotalItems] = React.useState(data.nodes.length || 0);
+  // Need to make header is sticky if first column is sticky otherwise the first header cell will not be sticky
+  const shouldHeaderBeSticky = isHeaderSticky || isFirstColumnSticky;
+  // Table Theme
+  const columnCount = getTableHeaderCellCount(children);
+  const firstColumnStickyHeaderCellCSS = isFirstColumnSticky
+    ? `
+  &:nth-of-type(1) {
+    left: 0 !important;
+    position: sticky !important;
+    z-index: ${firstColumnStickyHeaderFooterZIndex} !important;
+  }
+  ${
+    selectionType === 'multiple' &&
+    `&:nth-of-type(2) {
+    left: ${checkboxCellWidth}px !important;
+    position: sticky !important;
+    z-index: ${firstColumnStickyHeaderFooterZIndex} !important;
+  }
+  `
+  }`
+    : '';
+  const firstColumnStickyFooterCellCSS = isFirstColumnSticky
+    ? `
+  &:nth-of-type(1) {
+    left: 0 !important;
+    position: sticky !important;
+    z-index: ${firstColumnStickyHeaderFooterZIndex} !important;
+  }
```

Comment:
🤔 we are using strings here now? instead of objects?

Reply:
This is a hook given by react table library and it only accepts a string :(